### DISCUSSION TITLE
Undo/redo on entity store

### DISCUSSION THREAD

**Discussion by lilas78** [salesforce/akita#618] [Q&A] [answered]
I'm having an entity store, and I use StateHistoryPlugin to handle undo/redo. It works fine, except for one case :
  - an entity gets a new property 
  - What I expect :  when I do an undo on this entity, I expect to have the entity its in old state, so without the new property
  - What I have : after the undo, the entity still have this property

When debugging, I can see that the undo calls an "update" on the entity. As far as I know, an update makes a partial update, it means it changes the property that are passed as parameter to the update, but don't touch the other property. I would expect the undo to make a set instead of an update.

Can anybody explain me where I am wrong ?
```
entity = {
  name :"name",
  date: "date"
}
```
after user action : 
```
entity = {
  name :"name2",
  date : "date2",
  newProp : "prop"
}
```
after undo, I expect
```
entity = {
  name :"name",
  date : "date"
}
```
But I get 
```
entity = {
  name :"name",
  date : "date",
  newProp : "prop"
}
```

**Comment by NetanelBasal** [comment]
Can you reproduce it on stackblitz, please?

**Reply by lilas78** [reply]
[stackblitz](https://akita-books-dejkbp.stackblitz.io)

Click on edit button : this will add a comment on first book
Click on undo => the comment is not removed

**Comment by NetanelBasal** [comment, marked as answer]
I see the bug. When we call `update`, we merge the current entity with the new one. Please open an issue.
